src/utility.jl

```diff
@@ -74,3 +74,115 @@ function visualize_flow(method::ColorBased, flow::Array{SVector{2, Float64}, 2};
     return hsv
     
 end
+
+struct Point{T}
+    x::T
+    y::T
+end
+
+function check_flow(p::Point)
+    return !isnan(p.x) && !isnan(p.y) && abs(p.x) < 1e9 && abs(p.y) < 1e9
+end
+
+function check_flow(p::Array{Float64, 1})
+    return !isnan(p[1]) && !isnan(p[2]) && abs(p[1]) < 1e9 && abs(p[2]) < 1e9
+end
+
+"""
+```
+end_point_error(ground_truth_flow, estimated_flow)
+```
+
+Returns a 2-Dimensional array that matches the dimensions of the input flow vector and
+that depicts the end point error between the estimated flow and the ground truth flow.
+
+# Details
+If the estimated flow at a point is (u0, v0) and ground truth flow is (u1, v1), then 
+error will be sqrt[(u0 - u1)^2 + (v0 - v1)^2] at that point.
+
+# Arguments
+The flow parameters needs to be two-dimensional arrays of length-2 vectors (of type SVector) 
+which represent the displacement of each pixel.
+
+# Example
+
+Compute the end point error between two flows.
+```julia
+
+using ImageTracking
+
+result = end_point_error(ground_truth_flow, estimated_flow)
+
+imshow(result)
+```
+"""
+function end_point_error(ground_truth_flow::Array{SVector{2, Float64}, 2}, estimated_flow::Array{SVector{2, Float64}, 2})
+    result = Array{Float64, 2}(undef, size(estimated_flow)[1], size(estimated_flow)[2])
+    for i in 1:size(estimated_flow)[1]
+        for j in 1:size(estimated_flow)[2]
+            p1 = Point{Float64}(estimated_flow[i, j][1], estimated_flow[i, j][2])
+            p2 = Point{Float64}(ground_truth_flow[i, j][1], ground_truth_flow[i, j][2])
+
+            if check_flow(p1) && check_flow(p2) 
+                diff = Point{Float64}(p1.x - p2.x, p1.y - p2.y)
```

Comment:
Do we need to convert to a `Point{Float64}` here?  Why not just `diff = p1 - p2` ? Or even better, `δ = p1 - p2`.